src/cryptonote_basic/cryptonote_basic_impl.cpp

```diff
@@ -310,6 +310,42 @@ namespace cryptonote {
   bool operator ==(const cryptonote::block& a, const cryptonote::block& b) {
     return cryptonote::get_block_hash(a) == cryptonote::get_block_hash(b);
   }
+  //--------------------------------------------------------------------------------
+  int compare_hash32_reversed_nbits(const crypto::hash& ha, const crypto::hash& hb, unsigned int nbits)
+  {
+    const uint64_t* const va = reinterpret_cast<const uint64_t*>(ha.data);
+    const uint64_t* const vb = reinterpret_cast<const uint64_t*>(hb.data);
+
+    for (int n = 3; n >= 0 && nbits; --n)
+    {
+      const unsigned int msb_nbits = std::min((unsigned int) 64, nbits);
+      const uint64_t lsb_nbits_dropped = static_cast<uint64_t>(64 - msb_nbits);
+      const uint64_t van = SWAP64LE(va[n]) >> lsb_nbits_dropped;
+      const uint64_t vbn = SWAP64LE(vb[n]) >> lsb_nbits_dropped;
+      nbits -= msb_nbits;
+
+      if (van < vbn) return -1; else if (vbn > van) return 1;
+    }
+
+    return 0;
+  }
+
+  crypto::hash make_hash32_loose_template(unsigned int nbits, const crypto::hash& h)
+  {
+    crypto::hash res = h;
+    uint64_t* vh = reinterpret_cast<uint64_t*>(res.data);
+
+    for (int n = 3; n >= 0; --n)
+    {
+      const unsigned int msb_nbits = std::min((unsigned int) 64, nbits);
+      const uint64_t mask = msb_nbits ? (~((static_cast<uint64_t>(1) << (64 - msb_nbits)) - 1)) : 0;
```

Comment:
For future reference, `std::uint64_t(1)` does the same thing as the `static_cast` but with fewer characters.